plugins/obs-ffmpeg/obs-ffmpeg-output.c

```diff
@@ -877,15 +909,14 @@ static void receive_audio(void *param, struct audio_data *frame)
 	frame_size_bytes = (size_t)data->frame_size * data->audio_size;
 
 	for (size_t i = 0; i < data->audio_planes; i++)
-		circlebuf_push_back(&data->excess_frames[i], in.data[i],
+		circlebuf_push_back(&data->excess_frames[track_order][i], in.data[i],
 				in.frames * data->audio_size);
 
-	while (data->excess_frames[0].size >= frame_size_bytes) {
+	while (data->excess_frames[track_order][0].size >= frame_size_bytes) {
 		for (size_t i = 0; i < data->audio_planes; i++)
-			circlebuf_pop_front(&data->excess_frames[i],
-					data->samples[i], frame_size_bytes);
-
-		encode_audio(output, context, data->audio_size);
+			circlebuf_pop_front(&data->excess_frames[track_order][i],
```

Comment:
Style:  The `circlebuf_pop_front` lines are a bit too long, so they need to be broken up a bit differently.